engine/servermaster/jobmanager.go

```diff
@@ -431,5 +467,36 @@ func (jm *JobManagerImplV2) OnWorkerStatusUpdated(worker lib.WorkerHandle, newSt
 
 // CloseImpl implements lib.MasterImpl.CloseImpl
 func (jm *JobManagerImplV2) CloseImpl(ctx context.Context) error {
+	jm.notifier.Close()
 	return nil
 }
+
+// WatchJobStatuses returns a snapshot of job statuses followed by a stream
+// of job status changes.
+func (jm *JobManagerImplV2) WatchJobStatuses(
+	ctx context.Context,
+) (resManager.JobStatusesSnapshot, *notifier.Receiver[resManager.JobStatusChangeEvent], error) {
+	// We add an explicit deadline to make sure that
+	// any potential problem will not block the JobManager forever.
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
When `SubmitJob` is called, 
(1) Should we call `WatchJobStatuses` again? Or just add to the job snapshot?
(2) Should we notify the notifier?

Reply:
- We will postpone implementing notification of the `SubmitJob` event.
- The caller of `WatchJobStatuses` is told about any changes by the `receiver`. JobManager only needs the add a notification to the `notifier`.